proto/encode.go

```diff
@@ -59,6 +59,10 @@ func (e *RequiredNotSetError) Error() string {
 	return fmt.Sprintf("proto: required field %q not set", e.field)
 }
 
+func NewRequiredNotSetError(field string) *RequiredNotSetError {
```

Comment:
Please move this to encode_gogo.go
I try to gogoprotobuf's extra code as seperate as possible from the original goprotobuf code.